### PR TITLE
Distributed error reporting: Setting up the database that stores all the errors

### LINKED ISSUE #12244: Setup the database that stores all the errors raised during runtime
## Overview
This issues involves setting up the database that stores all the errors raised from the frontend and backend during runtime.

## Description and outcomes
- Add a new sqlite database in [sqlite_db_names.py](https://github.com/learningequality/kolibri/blob/80939fe789ed169504862415c4441d0ba37e932f/kolibri/deployment/default/sqlite_db_names.py#L14), with the name `errors`, to store all the errors that are traced from the frontend and the backend.
- Create an app called `errorsreports` to handle all error reporting related logic in `kolibri/core/`. Remember to add the new app to the list of `INSTALLED_APPS`.
- Add a [database router](https://github.com/learningequality/kolibri/blob/09297be74fa5239a1e0a5da218e13d2281a8ad18/kolibri/deployment/default/settings/base.py#L161-L165) for the `errors` database, and implement the database's routing behavior in `kolibri.core.errors.models.ErrorsRouter`.
- All error related reads and writes, regardless of the application they originate from, should be routed to the `errorsreports` app database.
- Run a migration to create the new sqlite database.

## Acceptance Criteria
- An app called `errorsreports` has been created in `kolibri/core/`
- Database migration has been executed and a new database `errors.sqlite3` has been created in `KOLIBRI_HOME`, in the default or specified location.
- All error related reads and writes, regardless of the application they originate from, are routed to the `errorsreports` app database.

## Assumptions and Dependencies
NA

## Scope
This shall cover:
- Creating the errors app
- Implementing the routing behavior for the errors database
- Running the migration to create the new database

It doesn’t include;
- Implementing the model that will store the reported errors

## Accessibility Requirements
NA

## Resources
- [ Multiple databases in Django](https://docs.djangoproject.com/en/3.2/topics/db/multi-db/)

### PR DESCRIPTION
**Pull request by thesujai**
## Summary

1. Create a new `errorreports` app and add it to INSTALLED_APPS
2. Add `errorreports` database name to the additional sqlite database
3. Create an ErrorReportsRouter to ensure all the reported errors are stored in a separate database -> errorreports.sqlite3

…

## References

Fixes #12244
…

## Reviewer guidance

I created a TestModel in errorreports app, which I then migrated and tested in a shell that the creation of objects with that model is written to errorreports.sqlite3 and not any other databases.
…

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Review by akolson — COMMENTED** [review]
Setup looks correct to me! manual tests checkout! @rtibbles any other thoughts?

Screenshot 2024-06-06 at 21 32 40

File: kolibri/core/errorreports/apps.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from django.apps import AppConfig
+
+
+class KolibriErrorConfig(AppConfig):
+    name = "kolibri.core.errorreports"
+    label = "errorreports"
+    verbose_name = "Kolibri ErrorReports"
+
+    def ready(self):
```
Comment (akolson) [inline kolibri/core/errorreports/apps.py]: There's probably no use case for this method. However if there is, then its worth changing its body from `...` to `pass` that is more explicit

**Review by akolson — CHANGES_REQUESTED** [review]
Hi @thesujai! Just 1 blocking change, but we should be good for a merge once update is made!

File: kolibri/core/errorreports/apps.py (hunk 1 above)
Comment (thesujai) [reply kolibri/core/errorreports/apps.py]: I thought both were the same thing!

Comment (thesujai) [reply kolibri/core/errorreports/apps.py]: Changed to pass!

Comment (akolson) [reply kolibri/core/errorreports/apps.py]: Pass is more explicit. ... could mean a host of things.
It goes back to my earlier question. Do we really need the ready method?

Comment (thesujai) [reply kolibri/core/errorreports/apps.py]: Not now, some apps inside core had it defined this way with pass. So I just copy-pasted

Comment (akolson) [reply kolibri/core/errorreports/apps.py]: I thinks it's best we remove it as it won't really be necessary.

**Review by akolson — APPROVED** [review]
LGTM. Thanks @thesujai!

**Pull request merged**